cpp/inc/bond/protocol/compact_binary.h

```diff
@@ -381,129 +381,231 @@ class CompactBinaryReader
     template <typename T>
     void Skip()
     {
-        Skip(get_type_id<T>::value);
+        SkipType<get_type_id<T>::value>();
     }
 
     template <typename T>
     void Skip(const bonded<T, CompactBinaryReader&>&)
     {
-        SkipComplex(bond::BT_STRUCT);
+        SkipType<bond::BT_STRUCT>();
     }
 
     void Skip(BondDataType type)
     {
-        switch (type)
+        SkipType(type);
+    }
+
+protected:
+    template <BondDataType T>
+    typename boost::enable_if_c<(T == BT_BOOL || T == BT_UINT8 || T == BT_INT8)>::type
+    SkipType(uint32_t size = 1)
+    {
+        _input.Skip(size * sizeof(uint8_t));
```

Comment:
We need to make sure that the multiplication doesn't overflow.

If this `size`--or any of the other `size` argument in the other overloads/specializations--come from the payload, we'll need to do a runtime check. (See https://github.com/Microsoft/bond/blob/master/cpp/inc/bond/core/detail/checked_add.h for helpers that we have for doing this on addition.)

If these are just other constants in the code, we can perform a case-by-case analysis and add a (static?) assert.

Reply:
Added #774.